Enemies

PathKun opened the pull request:
Summary: Put together enemy types
melee chase
ranged chase
tank
suicide bomber

Details: chase are based on navmesh, yet to implement navmesh on the rest
kamikaze still being worked on, a lot of clean up

Bug Fixes:

Deleted:

P1xelslayer committed:
Added the enemy

Adding the chase and folders for the enemies

P1xelslayer committed:
Merging all enemies

Merging the enemies
Tank
Chase
and SuicideBomber

Jeroldine committed:
Enemy tank behaviour (#53)

Summary:
Created an enemy tank game object with AI. Enemy also has a attack box

Details:
- Tank will chase enemy
- Rotate to face the direction it is travelling
- Debug message when player enters the enemy attack box

Bug Fixes:

Deleted:

PathKun committed:
Enemies kurt (#66)

Summary: Navmesh based movement for enemy

Details: Added melee chase and ranged chase with movement based on
navmesh

Bug Fixes:

Deleted:

cothon01 reviewed (changes requested):
There's two scripts that do the same thing. There should be only one of them:
- suicideBomberAI.cs
- Kamikaze behaivour.cs

Materials should not be in the Scripts folder.

cothon01 committed:
Created Floating HealthBar for Enemies (re-opened) (#70)

(Copied from #61. This one will be merged with the enemies branch
instead of main)

Created first version of floating enemy health bar

Summary: Created Health Bars for enemies that float above them in world
space

Details: Created slider to sit above enemies as healthbar, requires a
health, and a maxHealth value of enemies to display health on slider.

Bug Fixes: NULL

Deleted: NULL

---------

Co-authored-by: JacyWhitfield <129677131+JacyWhitfield@users.noreply.github.com>
Co-authored-by: soble14 <148224970+soble14@users.noreply.github.com>
Co-authored-by: Moseph90 <149262606+Moseph90@users.noreply.github.com>
Co-authored-by: Sneakydeaky <64993437+Sneakydeaky@users.noreply.github.com>
Co-authored-by: RosemaryBagels <148225175+RosemaryBagels@users.noreply.github.com>
Co-authored-by: Helcostr <15018549+Helcostr@users.noreply.github.com>
Co-authored-by: thatonenerdd <148226154+thatonenerdd@users.noreply.github.com>
Co-authored-by: CarlosTFS <148225676+CarlosTFS@users.noreply.github.com>
Co-authored-by: Seangao97 <Seangao97@gmail.com>
Co-authored-by: Mitchell Chesbro <mitchell.chesbro@hotmail.com>
Co-authored-by: EthanM5 <ethanjmcneill@hotmail.com>

cothon01 reviewed (dismissed):
Upon discussion with the enemies team, I will approve these changes for now, since the team requires a lot of scripts from main. The fixes will be done at a later stage.

cothon01 committed:
Merge branch 'main' into enemies

cothon01 reviewed (approved):
Conflicts resolved
